contracts/dao/voting/vesting-lp-vault/src/contract.rs

```diff
@@ -0,0 +1,343 @@
+        &config.atom_oracle_contract,
+        &query_msg,
+        height,
+    )?;
+    let usdc_power = get_voting_power(
+        deps,
+        &config.vesting_lp_contract,
+        &config.usdc_oracle_contract,
+        &query_msg,
+        height,
+    )?;
+
+    Ok(VotingPowerAtHeightResponse {
+        power: (atom_power + usdc_power).numerator().try_into()?,
+        height,
+    })
+}
+
+pub fn query_total_power_at_height(
+    deps: Deps,
+    env: Env,
+    height: Option<u64>,
+) -> StdResult<TotalPowerAtHeightResponse> {
+    let config = CONFIG.load(deps.storage)?;
+    let height = height.unwrap_or(env.block.height);
+
+    let query_msg = VestingLpQueryMsg::UnclaimedTotalAmountAtHeight { height };
+    let atom_power = get_voting_power(
+        deps,
+        &config.vesting_lp_contract,
+        &config.atom_oracle_contract,
+        &query_msg,
+        height,
+    )?;
+    let usdc_power = get_voting_power(
+        deps,
+        &config.vesting_lp_contract,
+        &config.usdc_oracle_contract,
+        &query_msg,
+        height,
+    )?;
+
+    Ok(TotalPowerAtHeightResponse {
+        power: (atom_power + usdc_power).numerator().try_into()?,
+        height,
+    })
+}
+
+pub fn query_info(deps: Deps) -> StdResult<Binary> {
+    let info = cw2::get_contract_version(deps.storage)?;
+    to_binary(&cwd_interface::voting::InfoResponse { info })
+}
+
+pub fn query_dao(deps: Deps) -> StdResult<Binary> {
+    let dao = DAO.load(deps.storage)?;
+    to_binary(&dao)
+}
+
+pub fn query_name(deps: Deps) -> StdResult<Binary> {
+    let config = CONFIG.load(deps.storage)?;
+    to_binary(&config.name)
+}
+
+pub fn query_description(deps: Deps) -> StdResult<Binary> {
+    let config = CONFIG.load(deps.storage)?;
+    to_binary(&config.description)
+}
+
+pub fn query_config(deps: Deps) -> StdResult<Binary> {
+    let config = CONFIG.load(deps.storage)?;
+    to_binary(&config)
+}
+
+pub fn query_list_bonders(
+    deps: Deps,
+    start_after: Option<String>,
+    limit: Option<u32>,
+    order_by: Option<OrderBy>,
+) -> StdResult<Binary> {
+    // TODO: this method does not adjust LP tokens amount to their corresponding voting power
+    let config = CONFIG.load(deps.storage)?;
+
+    let vesting_accounts: VestingAccountsResponse = deps.querier.query_wasm_smart(
+        config.vesting_lp_contract,
+        &VestingLpQueryMsg::VestingAccounts {
+            start_after,
+            limit,
+            order_by,
+        },
+    )?;
+
+    let bonders = vesting_accounts
+        .vesting_accounts
+        .into_iter()
+        .map(|vesting_account| BonderBalanceResponse {
+            address: vesting_account.address.into_string(),
+            balance: vesting_account
+                .info
+                .schedules
+                .into_iter()
+                .map(|vesting_schedule| vesting_schedule.start_point.amount)
+                .sum(), // TODO: probably deduct vesting_account.info.released_amount
```

Comment:
I think you can make use of this query here: https://github.com/neutron-org/neutron-tge-contracts/blob/main/packages/vesting-base/src/handlers.rs#L294

Reply:
again, it is just a draft for future, we better focus on query_voting_power queries